JonathanCoco/angular/controlling_users/index.html

```diff
@@ -0,0 +1,70 @@
+<html>
+<head>
+  <title></title>
+  <script src='bower_components/angular/angular.js'></script>
+  <script type='text/javascript'>
+    var myApp = angular.module('myApp', []);
+    myApp.controller('usersController', ['$scope', function ($scope){
+      // we don't need a factory for this demo, it wouldn't make a difference if we did.
+        $scope.sortType     = 'first_name'; // set the default sort type
+        $scope.sortReverse  = false;  // set the default sort order
+
+        $scope.users = [
+            {first_name:'Jon', last_name: 'Coco', language:'angular'},
+            {first_name:'Jennifer', last_name: 'Coco', language:'french'}
+        ];
+
+        $scope.addUser = function(firstName, lastName, favoriteLanguage){
+
+            user = { first_name: firstName, last_name: lastName, language: favoriteLanguage}
+            $scope.users.push(user);
+            console.log($scope.users);
+        }
+
+        $scope.deleteUser = function(index){
+
+            $scope.users.splice(index, 1);
+        }
+    }])
+  </script>
+</head>
+<body ng-app='myApp'>
+    <div ng-controller='usersController'>
+      <label>First Name</label>
+      <input type="text" ng-model = 'firstName'><br>
+
+      <label>Last Name</label>
+      <input type="text" ng-model = 'lastName'><br>
+
+      <label>Favorite Language</label>
+      <input type="text" ng-model = 'favoriteLanguage'><br>
+
+      <button ng-click = "addUser(firstName, lastName, favoriteLanguage);">Create User</button>
```

Comment:
You actually don't need to pass in the variables as arguments if you save your ng-models as a single object, i.e newUser.firstname, newUser.lastName, newUser.favoritelanguage.  You'll then have access to the object and it's keys through scope so when you call the addUser function in the controller, all you have to do is   $scope.users.push($scope.newUser).  Actually, you should still have access to the inputs the way you wrote it through scope so you can build out your user object inside the controller without having to pass them in as arguments.